AppCrashTracker/src/main/java/com/org/appcrashtracker/data/ExceptionData.java

```diff
@@ -0,0 +1,248 @@
+package com.org.appcrashtracker.data;
+
+import ohos.aafwk.ability.Ability;
+import ohos.bundle.BundleInfo;
+import ohos.global.resource.NotExistException;
+import ohos.global.resource.ResourceManager;
+import ohos.global.resource.WrongTypeException;
+import ohos.hiviewdfx.HiLog;
+import ohos.hiviewdfx.HiLogLabel;
+import ohos.system.DeviceInfo;
+import ohos.utils.zson.ZSONObject;
+import static com.example.appcrashtracker.ResourceTable.*;
+import static com.org.appcrashtracker.utils.MemoryUtils.*;
+import com.org.appcrashtracker.utils.ApplicationInfoUtils;
+import com.org.appcrashtracker.utils.BatteryUtils;
+import com.org.appcrashtracker.utils.NetworkUtils;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.Locale;
+/**
+ * Data class to handle all the data
+ * @author Raj Bhagat
+ * @version 1.0.0
+ */
+public class ExceptionData {
+
+    private final HiLogLabel label = new HiLogLabel(HiLog.LOG_APP, 0x00201, getClass().getName());
+    private boolean className = false;
+    private boolean message = false;
+    private boolean localizedMessage = false;
+    private boolean causes = false;
+    private boolean stackTraceBoolean = false;
+    private boolean brandName = false;
+    private boolean deviceName = false;
+    private boolean sdkVersion = false;
+    private boolean release = false;
+    private boolean height = false;
+    private boolean width = false;
+    private boolean appVersion = false;
+    private boolean tablet = false;
+    private boolean deviceOrientation = false;
+    private boolean screenLayout = false;
+    private boolean vmHeapSize = false;
+    private boolean allocatedVmSize = false;
+    private boolean vmMaxHeapSize = false;
+    private boolean vmFreeHeapSize = false;
+    private boolean nativeAllocatedSize = false;
+    private boolean batteryPercentage = false;
+    private boolean batteryCharging = false;
+    private boolean batteryChargingVia = false;
+    private boolean sdCardStatus = false;
+    private boolean internalMemorySize = false;
+    private boolean externalMemorySize = false;
+    private boolean internalFreeSpace = false;
+    private boolean externalFreeSpace = false;
+    private boolean packageName = false;
+    private boolean networkMode = false;
+    private boolean country = false;
+    private ZSONObject jsonObjectData = new ZSONObject();
+
+    /** Set the conditions of the details required */
+    public void setBoolean(Ability ability) throws NotExistException, WrongTypeException, IOException {
```

Comment:
No need for this method setBoolean. Booleans should be set in the constructor of ExceptionData class. Also, add Copyright text from the community copyright template.